lib/krane/kubernetes_resource.rb

```diff
@@ -237,7 +238,11 @@ def deploy_timed_out?
 
     # Expected values: :apply, :replace, :replace_force
     def deploy_method
-      :apply
+      if @definition.dig("metadata", "name").blank? && @definition.dig("metadata", "generateName").present?
```

Comment:
Can both be set?

Reply:
> GenerateName is an optional prefix, used by the server, to generate a unique name ONLY IF the Name field has not been provided. 

Tested it out, and if you have both, `generateName` is ignored.

Reply:
Nope